### PR TITLE
Ensure time picker input mode lays out correctly in RTL

### LINKED ISSUE #63427: Wrong hour/minute order in timePicker in RTL mode
Hour and Minute order are wrong in RTL mode. It is just reversed from left -> right to right -> left.
Hour view is same in both RTL and LTR meaning hour comes on the left and minute comes on the right.  
To reproduce the problem set ``locale`` in ``MaterialApp`` to an rtl locale (like ``Locale('fa')``); and call ``showTimePicker(context:context, initialTime: TimeOfDay.now());``
screenshot-2020-08-11_01 50 05 949

### PR DESCRIPTION
**Pull request by rami-a**
## Description

This is a follow-up to https://github.com/flutter/flutter/pull/62803 and ensures that the hours and minutes are laid out correctly for RTL for the time picker input mode (the previous PR only fixed the clock dial mode).

## Related Issues

Fixes #63427

## Tests

I added the following tests:

- a test to check the position of elements in various locales including an RTL one

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/time_picker.dart [hunk 1]
```diff
@@ -1414,58 +1414,64 @@ class _TimePickerInputState extends State<_TimePickerInput> {
...
+                          ExcludeSemantics(
+                            child: Text(
+                              MaterialLocalizations.of(context).timePickerHourLabel,
+                              style: theme.textTheme.caption,
+                              maxLines: 1,
+                              overflow: TextOverflow.ellipsis,
+                            ),
+                          ),
+                      ],
+                    )),
```
Comment (Renzo-Olivares) [inline packages/flutter/lib/src/material/time_picker.dart:1444]: nit: trailing comma

File: packages/flutter/lib/src/material/time_picker.dart [hunk 2]
```diff
@@ -1414,58 +1414,64 @@ class _TimePickerInputState extends State<_TimePickerInput> {
...
+                          ExcludeSemantics(
+                            child: Text(
+                              MaterialLocalizations.of(context).timePickerMinuteLabel,
+                              style: theme.textTheme.caption,
+                              maxLines: 1,
+                              overflow: TextOverflow.ellipsis,
+                            ),
+                          ),
+                      ],
+                    )),
```
Comment (Renzo-Olivares) [inline packages/flutter/lib/src/material/time_picker.dart:1471]: nit: trailing comma

File: packages/flutter/lib/src/material/time_picker.dart [hunk 3]
```diff
@@ -1414,58 +1414,64 @@ class _TimePickerInputState extends State<_TimePickerInput> {
...
+                      height: _kTimePickerHeaderControlHeight,
+                      child: _StringFragment(timeOfDayFormat: timeOfDayFormat),
                     ),
-                ],
-              )),
-              Container(
-                margin: const EdgeInsets.only(top: 8.0),
-                height: _kTimePickerHeaderControlHeight,
-                child: _StringFragment(timeOfDayFormat: timeOfDayFormat),
+                    Expanded(child: Column(
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/time_picker.dart:1450]: In the previous change, the Expand child was on the subsequent line. It's a little easier to read that way.

File: packages/flutter/lib/src/material/time_picker.dart [hunk 4]
```diff
@@ -1489,6 +1495,60 @@ class _TimePickerInputState extends State<_TimePickerInput> {
   }
 }
 
+class _HourTextField extends StatelessWidget {
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/time_picker.dart:1498]: It seems a little like overkill to factor out _HourTextField and _MinuteTextField, but OK.

**Review by HansMuller — APPROVED** [review]
LGTM

**Review by johnsonmh — APPROVED** [review]

File: packages/flutter/lib/src/material/time_picker.dart (hunk 1 above)
Comment (rami-a) [reply packages/flutter/lib/src/material/time_picker.dart]: Done

File: packages/flutter/lib/src/material/time_picker.dart (hunk 2 above)
Comment (rami-a) [reply packages/flutter/lib/src/material/time_picker.dart]: Done

File: packages/flutter/lib/src/material/time_picker.dart (hunk 3 above)
Comment (rami-a) [reply packages/flutter/lib/src/material/time_picker.dart]: Done

File: packages/flutter/lib/src/material/time_picker.dart (hunk 4 above)
Comment (rami-a) [reply packages/flutter/lib/src/material/time_picker.dart:1502]: yeah I would have preferred not to, but it made it easier to properly test this.

**Pull request merged**